caffeine/src/main/java/com/github/benmanes/caffeine/cache/LocalCacheFactory.java

```diff
@@ -118,10 +119,9 @@ final class MethodHandleBasedFactory implements LocalCacheFactory {
     final MethodHandle methodHandle;
 
     MethodHandleBasedFactory(Class<?> clazz) throws NoSuchMethodException, IllegalAccessException {
-      var constructor = LOOKUP.findConstructor(clazz, FACTORY);
-      this.methodHandle = constructor.asType(
-          constructor.type().changeReturnType(BoundedLocalCache.class));
+      this.methodHandle = LOOKUP.findConstructor(clazz, FACTORY).asType(FACTORY_CALL);
```

Comment:
To actually test that thins works, I ran a few tests here with the above VarHandle lookup for the `FACTORY` field commented out. By that I ensured this `MethodHandleBasedFactory` is always used.

Do we have Emma or similar to verify that all code is tested?

Reply:
Yes, once the CI passes then the jacoco coverage will be uploaded to [coveralls](https://coveralls.io/github/ben-manes/caffeine). That shows statement coverage.

I also setup [codecov](https://app.codecov.io/gh/ben-manes/caffeine) and [sonarqube](https://sonarcloud.io/project/overview?id=com.github.ben-manes.caffeine%3Acaffeine) which show branch coverage.

Reply:
It looks like it executes both variants (using `FACTORY` field and the dynamic one). I was just unsure, which one is used, because the fallback looks like not used under normal circumstances: https://coveralls.io/builds/61811130/source?filename=caffeine%2Fsrc%2Fmain%2Fjava%2Fcom%2Fgithub%2Fbenmanes%2Fcaffeine%2Fcache%2FLocalCacheFactory.java

Reply:
Looks like full coverage!

Reply:
Oh yes, I wrote a test to force the fallback just to be safe

Reply:
I think it was this one or else related tests

https://github.com/ben-manes/caffeine/blob/master/caffeine/src/test/java/com/github/benmanes/caffeine/cache/BoundedLocalCacheTest.java#L2574